ompi/mca/pml/ucx/pml_ucx.c

```diff
@@ -190,12 +190,22 @@ static int mca_pml_ucx_recv_worker_address(ompi_proc_t *proc,
 
 int mca_pml_ucx_open(void)
 {
+    unsigned major_version, minor_version, release_number;
     ucp_context_attr_t attr;
     ucp_params_t params;
     ucp_config_t *config;
     ucs_status_t status;
 
-    PML_UCX_VERBOSE(1, "mca_pml_ucx_open");
+    /* Check version */
+    ucp_get_version(&major_version, &minor_version, &release_number);
+    PML_UCX_VERBOSE(1, "mca_pml_ucx_open: UCX version %u.%u.%u",
+                    major_version, minor_version, release_number);
+
+    if ((major_version == 1) && (minor_version == 8)) {
+        /* disabled due to issue #8321 */
+        PML_UCX_VERBOSE(1, "UCX is disabled due to a known issue in version 1.8");
```

Comment:
```suggestion
        PML_UCX_VERBOSE(1, "The UCX PML is disabled because libucx v1.8 was loaded at run time, and there is a known catastrophic bug in UCX v1.8");
```